File: server/src/models/abstract_models/food_model.py
from mongoengine import *
from src.models.abstract_models.nutrition_base_model import NutritionBase

class Food(NutritionBase):
    foodName = StringField(required=True)
    servingSizeWeight = DecimalField(required=True, precision=2, min_value=0.01)
    servingSizeUnit = StringField(required=True)
    images = ImageField(db_alias="miao",msize=(250, 250, False), thumbnail_size=(150, 150, False))

    meta = {
        "abstract": True
    }

File: server/models/abstract_models/food.model.js
import { DetailedFoodCompositionSchema } from "./detailedFoodComposition.model.js";

export const FoodSchema = new DetailedFoodCompositionSchema();

FoodSchema.add({
  foodName: {
    type: String,
    required: true,
  },
  servingSizeWeight: {
    type: Number,
    required: true,
    default: 100,
    set: (w) => Math.round(w * 100) / 100,
    get: (w) => Math.round(w * 100) / 100,
  },
  servingSizeUnit: {
    type: String,
    required: true,
  },
  images: {
    type: String,
    required: true,
  },
});
